components/script/dom/xmlhttprequest.rs

```diff
@@ -1038,6 +1044,34 @@ impl XMLHttpRequest {
         encoding.decode(&self.response.borrow(), DecoderTrap::Replace).unwrap().to_owned()
     }
 
+    // https://xhr.spec.whatwg.org/#blob-response
+    fn blob_response(&self) -> Root<Blob> {
+        // Step 1
+        match self.response_blob.get() {
+            Some(response) => response,
```

Comment:
We could simplify this step as `if let Some(blob) = self.response_blob.get() { return blob; }`